packages/shared/src/contexts/NotificationsContext.tsx

```diff
@@ -0,0 +1,195 @@
+import React, {
+  ReactElement,
+  ReactNode,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import {
+  UseNotificationPermissionPopup,
+  useNotificationPermissionPopup,
+} from '../hooks/useNotificationPermissionPopup';
+import { BootApp } from '../lib/boot';
+import { isDevelopment, isTesting } from '../lib/constants';
+import AuthContext from './AuthContext';
+
+interface NotificationsContextData extends UseNotificationPermissionPopup {
+  unreadCount: number;
+  isInitialized: boolean;
+  isSubscribed: boolean;
+  isNotificationSupported: boolean;
+  isNotificationsReady?: boolean;
+  clearUnreadCount: () => void;
+  incrementUnreadCount: () => void;
+  onTogglePermission: () => Promise<NotificationPermission>;
+}
+
+const NotificationsContext =
+  React.createContext<NotificationsContextData>(null);
+
+export default NotificationsContext;
+
+export interface NotificationsContextProviderProps {
+  children: ReactNode;
+  unreadCount?: number;
+  isNotificationsReady?: boolean;
+  app: BootApp;
+}
+
+export const NotificationsContextProvider = ({
+  app,
+  children,
+  isNotificationsReady,
+  unreadCount = 0,
+}: NotificationsContextProviderProps): ReactElement => {
+  const isExtension = !!process.env.TARGET_BROWSER;
+  const [OneSignal, setOneSignal] = useState(null);
```

Comment:
This one was very tricky to see what it actually was.
Is there not a better default state for it?

Reply:
I'm not sure what possible default values we can use for this since this is coming from a package.

Reply:
Hmm alright, leave it as is, nothing urgent 🥳